### DISCUSSION TITLE
A recent change on github prohibits rendering files in a repo

### DISCUSSION THREAD

**Discussion by stef** [community/community#59704] [Repositories]
### Select Topic Area

Bug

### Body

If i chose a random repo, and click on a random file in that, all i get is a 

```
Error
Looks like something went wrong!
```

in my console i see this additional info:

```
Content Security Policy: The page’s settings blocked the loading of a resource at data: (“media-src”). $filename-i-wanted-to-look-at
```

in another brand new browser profile this works though. my guess is that maybe the firefox noscript plugin is being pedantic.

**Comment by ArdeshirV** [comment]
The error you encountered typically occurs when the Content Security Policy (CSP) defined for the website restricts or blocks the loading of a particular resource, in this case, a file you wanted to access. The data: protocol indicates that the content is being delivered inline within the HTML document itself.

Regarding the Firefox NoScript plugin, it is primarily designed to enhance security by selectively allowing or blocking scripts on webpages. If you suspect that the plugin may be causing the issue, you can try disabling it temporarily to see if it resolves the problem.

Here are a few steps you can take to troubleshoot and resolve the issue:

- Clear cache and cookies: Sometimes, cached data or stored cookies can interfere with the proper functioning of a webpage. Clearing them might help resolve the issue.
- Disable browser extensions: Temporarily disable any browser extensions or plugins, such as NoScript, that might affect the website's behavior. Then, check if the issue persists.
- Try a different browser: Use a different web browser altogether to see if the problem is specific to your current browser. This will help determine if the issue is related to the browser itself or something else.

**Comment by stef** [comment]
disabling noscript does not help solving this issue. 
opening the same page in a different browser profile does. 
clearing cache and cookies does help, but after logging in again i get the same problem.

(apparently opening the page in a different browserprofile helps, because in that profile i'm not logged in!)

**Comment by stef** [comment]
just checked in the different browser profile i am able to login and see files. this is all very strange, this is something new, never had this problem before, and i did not change anything, no new plugins. so i suppose something got updated in github that triggers this bug.

**Comment by stef** [comment]
i also tried disabling all addons in the problematic profile, still the same symptom.

**Comment by relistan** [comment]
@stef I've been experiencing the exact same issue in Firefox

**Comment by relistan** [comment]
It turns out to have been a corrupted sqlite file locally in my profile. I removed `/Users/<username>/Library/Application Support/Firefox/Profiles/<your profile>.default/ls-archive.sqlite` and this seems to have fixed it. I ran a loop over all the sqlite files in my profile with `PRAGMA integrity_check`, which is how I found it. You might have a different file that is corrupted. But since that file is used in migrating local storage, I think it might suddenly get used now because of something to do with the change at github.

**Comment by stef** [comment]
interesting, local storage issues is also something that is mentioned in my console

**Comment by stef** [comment]
hmmmm, i just tried this:

```shell
for a in **/*.sqlite; do echo -n "$a "; echo "PRAGMA integrity_check" | sqlite3 $a; done
```

and it returned all OK, so i guess it must be something else

**Comment by stef** [comment]
so it turns out that i have disabled `dom.storage.enabled` in about:config. if i enable that, github works again.

**Comment by stef** [comment]
i consider this a bug, it forces me to globally enable localstorage, which is a privacy issue. please explain why this is necessary, and please revert to the previous state where localstorage was not necessary to look at source files.

**Comment by stef** [comment]
so it turns out it was a bug, and it is now fixed as per: https://support.github.com/ticket/personal/0/2279094

**Comment by stef** [comment]
the problem is back again.

**Reply by echalone** [reply]
Same, I also have this problem on Firefox

**Reply by tehcog** [reply]
Same here, using firefox with ad blocks, etc.

**Reply by anerdwithaknife** [reply]
Same here. Started recently, Firefox on Windows 10, no addons installed at all. 

Works without issue in Chromium browsers like Brave/Edge/Chrome.
